### PR TITLE
feat: added loading spinner for g tag detector implementation

### PR DESCRIPTION
**Pull request by gnanammcp**
# Description

Figma Plugin / Add loading spinner to <g> tag detector

## Links

***required*** - *Links to resources related to this Pull Request*

### CODE REVIEW

**Review by bhanusengar — APPROVED** [review]

**Review by gauravchauhan3030 — APPROVED** [review]

File: packages/@momentum-design/figma-plugin-assets-export/src/ui/sections/tools/Tools.tsx [hunk 1]
```diff
@@ -12,21 +12,30 @@ interface Props {
...
+  gTagAssetUpdate: any
+  setGTagAssetUpdate: any
 }
 
-function Tools({ settings, selectedAssetSettingId, gTagAsset, setGTagAsset }: Props) {
+function Tools({ settings, selectedAssetSettingId, gTagAsset, setGTagAsset,
+  gTagAssetUpdate, setGTagAssetUpdate }: Props) {
   const gTag = '<g> Detector';
   const [show, setShow] = useState<boolean>(false);
+  const [loader, setLoader] = useState<boolean>(true);
```
Comment (cscheffauer) [inline packages/@momentum-design/figma-plugin-assets-export/src/ui/sections/tools/Tools.tsx:23]: naming this `isLoading` would be more meaningful. The name `loader` could be missunderstood

File: packages/@momentum-design/figma-plugin-assets-export/src/ui/sections/tools/Tools.tsx [hunk 2]
```diff
@@ -12,21 +12,30 @@ interface Props {
...
   const [gTagData, setGTagData] = useState<any[]>([]);
   const selectedAssetSetting = useMemo(() => settings?.assets[selectedAssetSettingId] || undefined, [settings]);
 
   useEffect(() => {
     setGTagData(gTagAsset);
-  }, [gTagAsset]);
+    if (show) {
+      setLoader(false);
+    }
+  }, [gTagAssetUpdate]);
```
Comment (cscheffauer) [inline packages/@momentum-design/figma-plugin-assets-export/src/ui/sections/tools/Tools.tsx:32]: the dependency array here is missing the "show" attribute

File: packages/@momentum-design/figma-plugin-assets-export/src/ui/sections/tools/Tools.tsx [hunk 3]
```diff
@@ -12,21 +12,30 @@ interface Props {
...
 
   const tagClick = () => {
-    gTagDetector(parent, selectedAssetSetting);
     setShow(true);
+    setTimeout(() => {
```
Comment (cscheffauer) [inline packages/@momentum-design/figma-plugin-assets-export/src/ui/sections/tools/Tools.tsx:36]: why is this timeout necessary?

File: packages/@momentum-design/figma-plugin-assets-export/src/ui/sections/tools/Tools.tsx [hunk 4]
```diff
@@ -35,7 +44,9 @@ function Tools({ settings, selectedAssetSettingId, gTagAsset, setGTagAsset }: Pr
 
   const modalClose = () => {
     setShow(false);
+    setLoader(true);
```
Comment (cscheffauer) [inline packages/@momentum-design/figma-plugin-assets-export/src/ui/sections/tools/Tools.tsx:47]: isLoading is set to true when closing the modal? that doesn't make sense

File: packages/@momentum-design/figma-plugin-assets-export/src/ui/hooks/useStateHandlers.ts [hunk 5]
```diff
@@ -1,13 +1,14 @@
...
 
 const useStateHandlers = () => {
   const [settings, setSettings] = useState<Settings>();
   const [assetChunks, setAssetChunks] = useState<AssetChunks>([[]]);
-  const [exportStatus, setExportStatus] = useState<ExportStatus>("");
-  const [storageStatus, setStorageStatus] = useState<StorageStatus>("");
+  const [exportStatus, setExportStatus] = useState<ExportStatus>('');
+  const [storageStatus, setStorageStatus] = useState<StorageStatus>('');
   const [gTagAsset, setGTagAsset] = useState<any[]>([]);
+  const [gTagAssetUpdate, setGTagAssetUpdate] = useState<boolean>(false);
```
Comment (cscheffauer) [inline packages/@momentum-design/figma-plugin-assets-export/src/ui/hooks/useStateHandlers.ts:11]: what does this `gTagAssetUpdate` do?

**Review by cscheffauer — CHANGES_REQUESTED** [review]

File: packages/@momentum-design/figma-plugin-assets-export/src/ui/sections/tools/Tools.tsx (hunk 1 above)
Comment (gnanammcp) [reply packages/@momentum-design/figma-plugin-assets-export/src/ui/sections/tools/Tools.tsx]: changed as isLoading.

File: packages/@momentum-design/figma-plugin-assets-export/src/ui/sections/tools/Tools.tsx (hunk 2 above)
Comment (gnanammcp) [reply packages/@momentum-design/figma-plugin-assets-export/src/ui/sections/tools/Tools.tsx]: removed this code

File: packages/@momentum-design/figma-plugin-assets-export/src/ui/hooks/useStateHandlers.ts (hunk 5 above)
Comment (gnanammcp) [reply packages/@momentum-design/figma-plugin-assets-export/src/ui/hooks/useStateHandlers.ts]: removed this code and changed

File: packages/@momentum-design/figma-plugin-assets-export/src/ui/sections/tools/Tools.tsx (hunk 3 above)
Comment (gnanammcp) [reply packages/@momentum-design/figma-plugin-assets-export/src/ui/sections/tools/Tools.tsx:32]: we are calling parent.postMessage and on message is async and await function so here I am using timeout.

Comment (cscheffauer) [reply packages/@momentum-design/figma-plugin-assets-export/src/ui/sections/tools/Tools.tsx:32]: async functions should be awaited! no timeout necessary.

Comment (cscheffauer) [reply packages/@momentum-design/figma-plugin-assets-export/src/ui/sections/tools/Tools.tsx:32]: checked - this is necessary. please add a comment to describe the reasoning

**Review by cscheffauer — APPROVED** [review]

**Review by cscheffauer — APPROVED** [review]

**Pull request merged**